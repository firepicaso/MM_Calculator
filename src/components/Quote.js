import React, { useState, useEffect } from 'react';

// API KEY WbOSWI37+aKQ9vX6ZHP+uA==QiFg4i75K9iPWB1U

function Quote() {
  const [quote, selectQuote] = useState({});
  const [loading, setLoading] = useState(true);
  const [error, showError] = useState('');

  useEffect(() => {
    fetch('https://api.api-ninjas.com/v1/quotes?category=knowledge', {
      headers: {
        'X-Api-Key': 'WbOSWI37+aKQ9vX6ZHP+uA==QiFg4i75K9iPWB1U',
      },
    })
      .then((res) => {
        if (!res.ok) {
          throw new Error('Response was inaccurate');
        }
        return res.json();
      })
      .then((data) => {
        const randomQuote = data[Math.floor(Math.random() * data.length)];
        selectQuote(randomQuote);
        setLoading(false);
      })
      .catch((error) => {
        showError(error.message);
        setLoading(false);
      });
  }, []);

  if (loading) {
    return <div className="loading">Loading the Quote</div>;
  }

  if (error) {
    return (
      <div className="error-message">
        Error:
        {error}
      </div>
    );
  }

  return (
    <div>
      <h1>Knowledge Quote of the Moment</h1>
      <p>{quote.quote}</p>
      <h4>
        ~
        {quote.author}
      </h4>
    </div>
  );
}

export default Quote;
