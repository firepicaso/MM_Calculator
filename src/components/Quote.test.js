import Quote from "./Quote";
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

test('testing Quote component', () => {
    const { container } = render(<Quote />);
    expect(container.firstChild).toMatchSnapshot();
});

test('simulating user interaction', () => {
    render(<Quote />);
    const quoteElement = screen.getByText('Loading the Quote');
    expect(quoteElement).toBeInTheDocument();
});