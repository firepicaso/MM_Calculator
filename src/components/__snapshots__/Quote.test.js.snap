// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Quote component 1`] = `
<div
  class="loading"
>
  Loading the Quote
</div>
`;
